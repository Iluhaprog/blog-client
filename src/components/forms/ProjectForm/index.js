import pf from './ProjectForm';
import { connect } from 'react-redux';
import { reduxForm } from 'redux-form';

const mapStateToProps = state => ({
    initialValues: {
        title: state.project.selected.title,
        description: state.project.selected.description,
        githubLink: state.project.selected.githubLink,
        projectLink: state.project.selected.projectLink,
    },
});

const mapDispatchToProps = dispatch => ({});

const ProjectForm = connect(mapStateToProps, mapDispatchToProps)(reduxForm({
    form: 'project',
    enableReinitialize: true,
})(pf));

export {
    ProjectForm,
};