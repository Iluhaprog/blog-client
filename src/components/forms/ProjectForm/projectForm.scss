.project-form {
    font-size: 16px;

    .drop-zone {
        min-width: 200px;
        min-height: 200px;
        margin-right: 50px;

        img.empty {
            margin-top: 70px;
        }
    }

    &__box {
        width: 100%;
    }

    .form__box, .textarea {
        max-width: 100%;
        width: 100%;
        margin-right: 30px;
    }

    .form__box:last-child {
        margin-right: 0;
    }

    .primary-button {
        min-width: 300px;
        margin: 0 auto;
    }
}