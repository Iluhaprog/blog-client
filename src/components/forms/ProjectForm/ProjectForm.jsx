import React from 'react';
import { Row } from '../../containers';
import { Field } from 'redux-form';
import { 
    Textarea, 
    LabeledInput, 
    DropZone 
} from '../../formElements';
import './projectForm.scss';
import { PrimaryButton } from '../../buttons';

export default props => {
    const { handleSubmit } = props;

    return (
        <form onSubmit={handleSubmit} className='project-form'>
            <div className='container'>
                <Row>
                    <Field
                        name='file'
                        component={DropZone}
                    />
                    <div className='project-form__box'>
                        <LabeledInput 
                            type='text'
                            name='title'
                            component='input'
                            label='Title'
                        />
                        <Textarea 
                            name='description'
                            placeholder='Description'
                        />
                    </div>
                </Row>
                <Row>
                    <LabeledInput 
                        type='text'
                        name='githubLink'
                        component='input'
                        label='Github'
                    />
                    <LabeledInput 
                        type='text'
                        name='projectLink'
                        component='input'
                        label='Project'
                    />
                </Row>
                <PrimaryButton 
                    text='Save'
                />
            </div>
        </form>
    );
};