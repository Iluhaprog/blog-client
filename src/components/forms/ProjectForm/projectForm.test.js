import React from 'react';
import { render, waitFor, fireEvent } from '@testing-library/react';
import { reduxForm } from 'redux-form';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import ProjectForm from './ProjectForm';

const mockStore = configureMockStore();
const decor = el => {
    const form = () => (el);
    return reduxForm({
        form: 'test',
    })(form);
}

describe('Test view of project form', () => {
    test('Should return view of project form', () => {
        const store = mockStore();
        const Pf = decor(<ProjectForm onSubmit={() => {}}/>)
        const { asFragment } = render(
            <Provider store={store}>
                <Pf />
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('Test events of project form', () => {
    test('Should handle submit for project form', async () => {
        const store = mockStore();
        const handleSubmit = jest.fn();
        const Pf = decor(<ProjectForm handleSubmit={handleSubmit}/>)
        const { container } = render(
            <Provider store={store}>
                <Pf />
            </Provider>
        );
        fireEvent.submit(container.querySelector('form'));
        await waitFor(() => {
            expect(handleSubmit).toHaveBeenCalledTimes(1);
        });
    });
});