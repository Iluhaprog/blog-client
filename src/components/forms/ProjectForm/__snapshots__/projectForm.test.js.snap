// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view of project form Should return view of project form 1`] = `
<DocumentFragment>
  <form
    class="project-form"
  >
    <div
      class="container"
    >
      <div
        class="row   "
      >
        <div
          class="drop-zone "
        >
          <img
            alt="empty"
            class="empty"
            src="picture.svg"
          />
          <input
            type="file"
          />
        </div>
        <div
          class="project-form__box"
        >
          <div
            class="form__box row-reverse"
          >
            <input
              name="title"
              type="text"
              value=""
            />
            <label
              for="title"
            >
              Title
            </label>
          </div>
          <textarea
            class="textarea"
            name="description"
            placeholder="Description"
          />
        </div>
      </div>
      <div
        class="row   "
      >
        <div
          class="form__box row-reverse"
        >
          <input
            name="githubLink"
            type="text"
            value=""
          />
          <label
            for="githubLink"
          >
            Github
          </label>
        </div>
        <div
          class="form__box row-reverse"
        >
          <input
            name="projectLink"
            type="text"
            value=""
          />
          <label
            for="projectLink"
          >
            Project
          </label>
        </div>
      </div>
      <button
        class="button default-button primary-button"
      >
        Save
      </button>
    </div>
  </form>
</DocumentFragment>
`;
